testbeam_analysis/tools/plot_utils.py

```diff
@@ -91,19 +92,19 @@ def plot_masked_pixels(input_mask_file, pixel_size=None, dut_name=None, output_p
         ax.set_ylim(-0.5, occupancy.shape[0] - 0.5)
         ax.set_xlabel("Column")
         ax.set_ylabel("Row")
-        output_pdf.savefig()
+        output_pdf.savefig(fig)
 
-        plt.figure()
-        ax = plt.subplot(111)
+        fig = Figure()
+        _ = FigureCanvas(fig)
+        ax = fig.add_subplot(111)
         ax.set_title('%s' % (dut_name, ))
         ax.imshow(occupancy, aspect=aspect, cmap=cmap, interpolation='none', origin='lower', clim=(0, c_max))
     #     np.ma.filled(occupancy, fill_value=0)
         ax.set_xlim(-0.5, occupancy.shape[1] - 0.5)
         ax.set_ylim(-0.5, occupancy.shape[0] - 0.5)
         ax.set_xlabel("Column")
         ax.set_ylabel("Row")
-        output_pdf.savefig()
-        plt.close()
+        output_pdf.savefig(fig)
```

Comment:
Can we make storing to pdf optional, since we might just use the plot for the gui? But we do not have to do this now...